### PR TITLE
feat(store-devtools): change connectOutsideZone to be 'true' by default

### LINKED ISSUE #4093: Store Devtools: Set default for connectOutsideZone to true
### Which @ngrx/* package(s) are relevant/related to the feature request?

store-devtools

### Information

We introduced the `connectOutsideZone` flag for Store Devtools to connect to the Redux Devtools outside the Angular zone to reduce the number of change detection calls being triggered.

It is currently set to `false` by default. This change would flip the flag to `true` by default and update the associated documentation to reflect that.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by va-stefanek**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Closes #4093 

## Does this PR introduce a breaking change?

```
[x] Yes
[] No
```

### CODE REVIEW

File: modules/store-devtools/src/devtools.ts [hunk 1]
```diff
@@ -41,6 +48,7 @@ export class StoreDevtools implements Observer<any>, OnDestroy {
   public dispatcher: ActionsSubject;
   public liftedState: Observable<LiftedState>;
   public state: StateObservable;
+  zone = inject(NgZone);
```
Comment (timdeschryver) [inline modules/store-devtools/src/devtools.ts:51]: Why is this added?

**Review by timdeschryver — COMMENTED** [review]
From an API perspective I don't like having booleans default to `true`.
Since we're going to provide a migration, could we also rename this property?
Maybe `connectInsideZone`, with the default value of `false`.

The migration will add `connectInsideZone: true` to the config.

**Comment by brandonroberts** [comment]
> cc @brandonroberts , @markostanimirovic what do you think of a rename?

I think your points are reasonable, and the upgrade migration would cover existing projects.

**Comment by timdeschryver** [comment]
@va-stefanek sorry, but can you rename the property to `connectInsideZone` (with the default value of `false`)?

**Comment by timdeschryver** [comment]
`connectInZone` also works for me 👍

File: projects/ngrx.io/content/guide/store-devtools/index.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ bootstrapApplication(AppComponent, {
       autoPause: true, // Pauses recording actions and state changes when the extension window is not open
       trace: false, //  If set to true, will include stack trace for every dispatched action, so you can see it in trace tab jumping directly to that part of code
       traceLimit: 75, // maximum stack trace frames to be stored (in case trace option was provided as true)
-      connectOutsideZone: true // If set to true, the connection is established outside the Angular zone for better performance
+      connectInZone: false // If set false true, the connection is established outside the Angular zone for better performance
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store-devtools/index.md:59]:
```suggestion
      connectInZone: true // If set to true, the connection is established within the Angular zone
```

File: projects/ngrx.io/content/guide/store-devtools/index.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ import { StoreDevtoolsModule } from '@ngrx/store-devtools';
       autoPause: true, // Pauses recording actions and state changes when the extension window is not open
       trace: false, //  If set to true, will include stack trace for every dispatched action, so you can see it in trace tab jumping directly to that part of code
       traceLimit: 75, // maximum stack trace frames to be stored (in case trace option was provided as true)
-      connectOutsideZone: true // If set to true, the connection is established outside the Angular zone for better performance
+      connectInZone: false // If set to false, the connection is established outside the Angular zone for better performance
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store-devtools/index.md:31]:
```suggestion
      connectInZone: true // If set to true, the connection is established within the Angular zone
```

**Review by brandonroberts — APPROVED** [review]

**Review by timdeschryver — APPROVED** [review]
Thanks @va-stefanek
